cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -216,6 +222,8 @@ void SparsePoseGraph::ComputeConstraintsForScan(
   for (const mapping::Submap* submap : insertion_submaps) {
     const int submap_index = GetSubmapIndex(submap);
     CHECK(!submap_states_[submap_index].finished);
+    // however probability_grid might be final, so the last ray cast is drawn,
```

Comment:
Again just a remark, where my expectation from reading the code turned out wrong after a closer look.
For some reason I don't remember I wanted to point out, that the `CHECK` above does not imply that `submap.probability_grid` is still `Null`. So a not yet finished Submap might be in the process of finishing, where just the constraints to this last scan are still missing.